docs/cudf/source/developer_guide/contributing_guide.md

```diff
@@ -21,7 +21,6 @@ Specifically, cuDF uses the following tools:
 - [`mypy`](http://mypy-lang.org/) performs static type checking.
   In conjunction with [type hints](https://docs.python.org/3/library/typing.html),
   `mypy` can help catch various bugs that are otherwise difficult to find.
-- [`pydocstyle`](https://github.com/PyCQA/pydocstyle/) lints docstring style.
```

Comment:
Can we remove any other hooks that were replaced by ruff? (Check black, isort, …)

Reply:
Isort not yet, but black can go.

Reply:
Removed `black` from this list. I have a branch with `isort` but it's a little more complicated since `cudf_dask` `custreamz` and `cudf_kafka` have slightly difference isort configs

Reply:
> but it's a little more complicated since cudf_dask custreamz and cudf_kafka have slightly difference isort configs

I think you want to do this by having common ruff configs in the top-level `pyproject.toml` and then specific ones in each package:
```
[tool.ruff]
extend = "../../pyproject.toml"
[tool.ruff.lint.isort]
section-order = [
  "future",
  "standard-library",
  "third-party",
  "dask",
  "rapids",
  "first-party",
  "local-folder",
]
known-first-party = [
  "cudf",
]
combine-as-imports = true
order-by-type = true

[tool.ruff.lint.isort.sections]
"dask" = ["dask", "distributed", "dask_cuda"]
"rapids" = ["rmm"]
```
(for the cudf one, for example)

Reply:
Yup exactly, also documented in #14882